tests/bluetooth/audio/cap_commander/src/test_distribute_broadcast_code.c

```diff
@@ -0,0 +1,254 @@
+/* test_distribute_broadcast_code.c - unit test for distribute broadcast code */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/fff.h>
+
+#include "bluetooth.h"
+#include "cap_commander.h"
+#include "conn.h"
+#include "expects_util.h"
+#include "cap_mocks.h"
+#include "test_common.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_distribute_broadcast_code, CONFIG_BT_CAP_COMMANDER_LOG_LEVEL);
+
+#define FFF_GLOBALS
+
+struct cap_commander_test_distribute_broadcast_code_fixture {
+	struct bt_conn conns[CONFIG_BT_MAX_CONN];
+
+	struct bt_bap_bass_subgroup subgroups[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS];
+	struct bt_cap_commander_broadcast_reception_start_member_param
+		start_member_params[CONFIG_BT_MAX_CONN];
+	struct bt_cap_commander_broadcast_reception_start_param start_param;
+	struct bt_cap_commander_distribute_broadcast_code_member_param
+		broadcast_code_member_params[CONFIG_BT_MAX_CONN];
+	struct bt_cap_commander_distribute_broadcast_code_param distribute_broadcast_code_param;
+	struct bt_bap_broadcast_assistant_cb broadcast_assistant_cb;
+};
+
+static void cap_commander_test_distribute_broadcast_code_fixture_init(
+	struct cap_commander_test_distribute_broadcast_code_fixture *fixture)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		test_conn_init(&fixture->conns[i]);
+		fixture->conns[i].index = i;
+	}
+
+	fixture->distribute_broadcast_code_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	fixture->distribute_broadcast_code_param.param = fixture->broadcast_code_member_params;
+	fixture->distribute_broadcast_code_param.count =
+		ARRAY_SIZE(fixture->broadcast_code_member_params);
+	memcpy(fixture->distribute_broadcast_code_param.broadcast_code, BROADCAST_CODE,
+	       sizeof(BROADCAST_CODE));
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->broadcast_code_member_params); i++) {
+		fixture->broadcast_code_member_params[i].member.member = &fixture->conns[i];
+		fixture->broadcast_code_member_params[i].src_id = RANDOM_SRC_ID;
+	}
```

Comment:
Hmm, why do we need that check anyways? Since we are not storing the value in the stack anyways, and do not have a `src_id` check in the actual implementation, I don't see why the check makes sense here in the mock.

Reply:
It ensures that we copied the paramaters from `bt_cap_commander_distribute_broadcast_code`  correctly into the call to `bt_bap_broadcast_assistant_set_broadcast_code`